exporters/elasticsearch/src/es_log_exporter.cc

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/exporters/elasticsearch/es_log_exporter.h"
+#include "opentelemetry/exporters/elasticsearch/es_log_recordable.h"
+
+namespace nostd       = opentelemetry::nostd;
+namespace sdklogs     = opentelemetry::sdk::logs;
+namespace http_client = opentelemetry::ext::http::client;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * This class handles the response message from the Elasticsearch request
+ */
+class ResponseHandler : public http_client::EventHandler
+{
+public:
+  /**
+   * Automatically called when the response is received, store the body into a string and notify any
+   * threads blocked on this result
+   */
+  void OnResponse(http_client::Response &response) noexcept override
+  {
+    // Store the body of the request
+    body_ = std::string(response.GetBody().begin(), response.GetBody().end());
+
+    // Set the response_received_ flag to true and notify any threads waiting on this result
+    response_received_ = true;
+    cv_.notify_all();
+  }
+
+  /**
+   * A method the user calls to block their thread until the response is received, or the timeout is
+   * exceeded.
+   */
+  bool waitForResponse(unsigned int timeoutSec = 1)
+  {
+    std::mutex mutex_;
```

Comment:
nit: is this mutex useful if it is only defined in scope of this function?

Reply:
hmm it's needed for the condition variable `wait_for` method to work, but looks like traces does it [more condensed](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/src/trace/batch_span_processor.cc#L92) so good catch, I'll do it that way too

Reply:
jfyi - this code would become much simplified once we have synchronous request PR merged, as we won't need to wait for response anymore  : https://github.com/open-telemetry/opentelemetry-cpp/pull/448

This is how it would look like ( ref- zipkin exporter ) : https://github.com/open-telemetry/opentelemetry-cpp/blob/1039f4c2b947f3b3fd0880457ec77743fd7a80e7/exporters/zipkin/src/zipkin_exporter.cc#L62

Reply:
Alright alright, I'll add a lock_guard around body_ and response_received_ so they can't be read and written at the same time. Yea, this is probably going to be replaced by Lalit's synchronous request, but for the sake of getting this merged ASAP we can always change it in a later issue/PR.